### PR TITLE
feat(template): support pug template

### PR DESCRIPTION
**Pull request by yanskun**
## Types of changes

- [x] New feature (a non-breaking change which adds functionality)

## Description

create-nuxt-app is essential for creating Nuxt Project and I love it.
That's why I use this over and over again.

I prefer Pug over HTML because it's easier to read.
Also, I feel that there are many such friends. (#378)

After creating-nuxt-app, every time pug is installed and the page file is changed to `<template lang="pug">`.

I wanted to solve this, so I tried to fix it this time!
thank you

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

**Comment by yasudanaoya** [comment]
Someone was thinking the same thing!

https://github.com/nuxt/create-nuxt-app/issues/378

**Comment by yanskun** [comment]
@Atinux (I didn't know who to mention)
I love Nuxt, and  love Pug just as much.

I feel this is not actively developed.
However, until Nuxt3 is officially released, it will be used.

How can I get you to review.

**Comment by Atinux** [comment]
Thank you for your help @yanskun 

I added Clark as review since he has been working a lot on this project.

File: packages/cna-template/template/nuxt/pages/index.vue [hunk 1]
```diff
@@ -1,6 +1,12 @@
+<%_ if (pug) { _%>
```
Comment (clarkdo) [inline packages/cna-template/template/nuxt/pages/index.vue:1]: I think we can put pug templates into a separate dir like `/cna-template/template/frameworks/pug` for avoiding changing existing default template and also having more pug template files.

Comment (yanskun) [reply packages/cna-template/template/nuxt/pages/index.vue]: @clarkdo 
Thank you for your review.

The TypeScript branched IF in this file, so I thought this file would be fine.

I fixed to add `/cna-template/template/frameworks/pug` folder!

File: packages/cna-template/template/frameworks/pug/pages/index.vue [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<template lang="pug">
+  tutorial
+</template>
+
+<script>
+import Vue from 'vue'
+
+export default Vue.extend({
+  name: 'IndexPage'
+})
```
Comment (clarkdo) [inline packages/cna-template/template/frameworks/pug/pages/index.vue:10]: Let's add the end tag

**Review by clarkdo — APPROVED** [review]
Nice!

**Pull request merged**